x-pack/plugins/maps/public/shared/layers/tile_layer.js

```diff
@@ -43,6 +43,12 @@ export class TileLayer extends AbstractLayer {
     startLoading(SOURCE_DATA_ID_ORIGIN, requestToken, dataFilters);
     try {
       const url = await this._source.getUrlTemplate();
+      // Unparseable URLs can crash mapbox - ensure URL is valid
+      try {
+        new URL(url);
+      } catch (error) {
```

Comment:
Here I tihnk we should call `onLoadError(SOURCE_DATA_ID_ORIGIN, request, error.message)` and then `return`, iso throwing.

Alternatively, you can move this validation to `AbstractTMSSource#getUrlTemplate` and just throw there.